### ISSUE TITLE
Feature request : Make "go through" the default mode for BA/Infantry

### ISSUE THREAD

**Issue by Question2** [MegaMek/megamek#649]
Not sure if this is possible, but right now every unit defaults to climb up when you click on a building hex, and having it default to "go through" for BA/Infantry would save a LOT of time, since thats the most common mode when a player wants to move into a building hex with BA/infantry anyway.

**Comment by beerockxs** [comment]
That would also mean that we default to not move over a bridge, but move next to it on the ground.

**Comment by arlith** [comment]
This is a known issue that I've been thinking about for a while.  I'm not sure if there's another ticket related to it (I thought there was, but I don't recall what it is).  I've been thinking about having different default modes for different unit types.  I haven't really thought out how this would work, but it could solve some issues.  The other approach might be to have different climb modes for bridges vs buildings.

**Comment by dericpage** [comment]
I suspect that having climb mode default differently by unit type will lead to more confusion than defaulting climb mode by terrain type.

**Comment by Dylan-M** [comment]
I tend to agree with @dericpage. Go through for buildings, climb up for bridges (to go over) if you're walking onto a hex where the bridge starts, and climb under (for walking beneath) of bridges when you're crossing their path on the lower elevation terrain beneath them.

**Comment by arlith** [comment]
Maybe we should display a unit's current climb mode(s?) somewhere in the unit overview.  I mentioned this on slack the other day, and @HammerGS has mentioned it in the past, but I feel like climb mode can be confusing to users, but I can't really think of better ways to communicate it to users.  It could potentially be display on the general tab or something.

**Comment by arlith** [comment]
On the other hand, if we support different defaults for different unit types, we may likely be able to get to a point where users won't need to change climb mode in 95% of cases.

**Comment by Question2** [comment]
@arlith  : I noticed that when you use the climb button to switch climb modes, it shows which mode you currently have selected in blue text over the unit, but when you just click on a building hex, it doesn't. So a new user is probably not going to know that clicking on a building hex defaults to climbing it.

**Comment by arlith** [comment]
I was going to mention too that you can change the unit default in client settings.  If you would prefer go-through to be the default for all units, you can configure that now.

**Comment by BLOODWOLF333** [comment]
Alright we changed climb mode because people were making bug reports about trying to walk over bridges and the game instead making them "go under" and fall to their deaths.  And now we have VTOLs trying to fly through buildings and naval vessels trying to climb bridges.  

Anything that flies = default High climb mode
All ground units = default Low climb mode, except if terrain is bridge then will switch to High
Naval = default Low for everything.

If that is somehow confusing and not intuitive then it would require a more complicated route of making new options in the client settings to push it off on the player to decide how they want it setup.  Right now the High setting for all units is the lesser of the evils.

**Comment by IllianiBird** [comment]
Honestly, we should just make Go Thru the default for everything. How often do you wanna Kool Aid through a building vs. play King Kong?

**Comment by HammerGS** [comment]
This is very outdated and was completely reworked as part of the climbing work. This would be a graphical error seperate from the intend of this issue.

**Comment by gsparks3** [comment]
> Honestly, we should just make Go Thru the default for everything. How often do you wanna Kool Aid through a building vs. play King Kong?

IIRC, the primary reason is that Go Through causes units to attempt to move _under_ bridges, even when approaching them from a connected road. This can be fixed by swapping to Climb Up mode, but it's somewhat unintuitive to do so when the bridge is level with the terrain you're approaching it from.

If the automation was capable of distinguishing between bridges and non-bridge buildings and using different defaults for each, that might help.
